packages/ui-filecoin/src/components/Network/NetworkSelector.tsx

```diff
@@ -0,0 +1,46 @@
+'use client'
+
+import { GlobeIcon } from '@phosphor-icons/react/dist/ssr'
+
+import { Listbox } from '../Listbox'
+
+import { useNetwork } from './NetworkProvider'
+import type { Network } from './types'
+
+type NetworkOption = {
+  id: Network
+  label: string
+}
+
+const networkOptions = [
+  {
+    id: 'calibration',
+    label: 'Calibration',
+  },
+  {
+    id: 'mainnet',
+    label: 'Mainnet',
+  },
+] satisfies Array<NetworkOption>
+
+type NetworkSelectorProps = {
+  options?: Array<NetworkOption>
+}
+
+export function NetworkSelector({
+  options = networkOptions,
```

Comment:
Should custom options be allowed? `Network` type seems fixed to `'calibration' | 'mainnet'`

Custom options could have invalid network IDs.

Also maybe `networks` instead of `options`?

```ts
// Either don't allow custom options:
export function NetworkSelector() {
  // No options prop, always use networkOptions
}

// OR make it more explicit:
type NetworkSelectorProps = {
  /** Override default networks. All IDs must be valid Network values. */
  networks?: ReadonlyArray<NetworkOption>
}
```